src/components/ProjectSettingsTabs/VariablesTab/transformers.ts

```diff
@@ -0,0 +1,43 @@
+import { ProjectVarsInput } from "gql/generated/types";
+import { FormToGqlFunction, GqlToFormFunction } from "../types";
+import { FormState } from "./types";
+
+export const gqlToForm: GqlToFormFunction = (data): FormState => {
+  if (!data) return null;
+
+  const {
+    vars: { privateVars, vars },
+  } = data;
+
+  return {
+    vars: Object.entries(vars).map(([varName, varValue]) => ({
+      varName,
+      varValue,
+      isPrivate: privateVars.includes(varName),
+    })),
+  };
+};
+
+export const formToGql: FormToGqlFunction = (
+  { vars: varsData }: FormState,
+  id
+) => {
+  const vars: ProjectVarsInput = {
+    vars: {} as ProjectVarsInput["vars"],
+    privateVarsList: [] as ProjectVarsInput["privateVarsList"],
+  };
+
+  varsData.forEach(({ varName, varValue, isPrivate }) => {
+    if (!varName) return;
+
+    if (isPrivate) {
+      vars.privateVarsList.push(varName);
+    }
+    vars.vars[varName] = varValue;
+  });
+
+  return {
+    projectRef: { id },
+    vars,
+  };
+};
```

Comment:
```suggestion
export const formToGql: FormToGqlFunction = (
  { vars: varsData }: FormState,
  id
) => {
  const vars = varsData.reduce(
    (acc, { varName, varValue, isPrivate }) => {
      if (!varName) return acc;
      if (isPrivate) {
        acc.privateVarsList.push(varName);
      }
      acc.vars[varName] = varValue;
      return acc;
    },
    {
      vars: {},
      privateVarsList: [],
    }
  );

  return {
    projectRef: { id },
    vars,
  };
};
```
![image](https://user-images.githubusercontent.com/4605522/146607932-4847629a-5f92-434b-ac42-9ea99da7bf5e.png)

lets try to minimize our use of type assertions since we lose  the type safety features typescript provides.